src/main/java/com/rocketden/tester/service/RunnerService.java

```
@@ -2,12 +2,26 @@
 
 import com.rocketden.tester.dto.RunDto;
 import com.rocketden.tester.dto.RunRequest;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class RunnerService {
 
+    private final SetupService setupService;
+    private final DockerService dockerService;
+
+    @Autowired
+    public RunnerService(SetupService setupService, DockerService dockerService) {
+        this.setupService = setupService;
+        this.dockerService = dockerService;
+    }
+
     public RunDto run(RunRequest request) {
-        return null;
+        String folder = setupService.createTempFolder(request);
+        RunDto output = dockerService.spawnAndRun(folder);
```

If `output` equals `null`, should we return an error instead of just `null`? I'm uncertain what this may look like, maybe `status` could equal a special value, but I figure for the future we may need better error reporting.

---

Yeah I think it it's null then it would make sense to throw an error - I updated the `spawnAndRun` method to return null in the catch block rather than at the end of the method so now if it ever returns null it should also print an error message (not the same as throwing an error but my logic is that we can do that once we add in the ApiError stuff which wasn't included in this PR)

---

That works by me!